### ISSUE TITLE
Default value for InitVar fields on dataclasses

### ISSUE THREAD

**Issue by roeap** [microsoft/pyright#174]
**Describe the bug**
This is probably not a bug, but rather due to the specific behavior of the dataclass implementation. The mechanism for dataclasses to recognize `InitVar` types seems to inspect the type of the field. As a consequence the respective field has a type of `InitVar[T]`, while a default value provided should have type `T`

**To Reproduce**
I am using the backport of `dataclass` to python 3.6. When defining a simple dataclass such as: 
```python
@dataclass
class SomeClass:
    i: int
    j: InitVar[int] = 1

    def __init__(self, j):
        self.i = j
```

also included in the project are the typestubs for dataclasses with teh specific line:
```python
class InitVar(Generic[_T]): ...
```
pyright will prompt, that a field of type `InitVar[int]` cannot be assigned a value of type `int`

**Expected behavior**
Pyright should recognize that the field will be a field of type `int`, if that case, or otherwise allow for an exception. 

**Screenshots or Code**
see repro

**VS Code extension or command-line**
VS Code extension Version 1.0.47

**Additional context**
thanks for this great tool!

**Comment by erictraut** [comment]
Thanks for the bug report. I didn't know about InitVar until you mentioned it. After reading the documentation in PEP 557 my only thought is "boy, that's a poor design choice". I'll think about how to implement this in pyright.

**Comment by erictraut** [comment]
Thanks again for the bug report. I've implemented this in a minimal manner for now. It now properly handles the InitVar member type and includes the member in the initializer and constructor methods, but it doesn't currently report an error if an attempt is made to directly access the InitVar member. Adding that check required more in-depth modifications. I figured it was more important to eliminate the false positive error first.

This is now implemented in version 1.0.50, which I just published.
